### ISSUE TITLE
odd failure creating SmoothingSplineLog

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#74]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/74) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

```python
import mass
import numpy as np

# once fixed modify test_calibration.py
# code at line 43 can be removed
# code at line 99 can be uncommented

# using PDB I can isolate the failure to this call
# but oddly, this call works fine when made in isolation
# so I put it first, so the file runs until the crash, then there is no more code
ph= np.array([  604.9186911 ,   658.11682861,   710.25965219,   761.46157549,
         811.81613372,   861.40110546,   910.28210151,   958.51515166,
        1006.14861411])
e=  np.array([ 3000.        ,  3333.33333333,  3666.66666667,  4000.        ,
        4333.33333333,  4666.66666667,  5000.        ,  5333.33333333,
        5666.66666667])
de= np.array([ 0.01,  0.01,  0.01,  0.01,  0.01,  0.01,  0.01,  0.01,  0.01])
dph=np.array([ 0.60491869,  0.65811683,  0.71025965,  0.76146158,  0.81181613,
        0.86140111,  0.9102821 ,  0.95851515,  1.00614861])
a=mass.mathstat.interpolate.SmoothingSplineLog(ph,e,de,dph)

cal = mass.calibration.energy_calibration.EnergyCalibration()
for energy in np.linspace(3000,6000,10):
    ph = energy**0.8
    cal.add_cal_point(ph, energy)
cal.drop_one_errors() # this causes error
```

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Fixes #74: linear algebra error bug in SmoothingSpline.

Also restored tests in test_calibration that were failing 
and which Galen had to remove in c89c1c27d7eea9a.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Change fits' BG model so that it is never negative.

Now the model is still BG = BG(0bin) + bin# * BGslope, but it
goes to 0 when that formula is negative. Now we can allow the BG(0bin)
value to go negative safely. This fixes #74.
